src/PiggyCustomEnchants/EventListener.php

```diff
@@ -1093,117 +1128,119 @@ public function checkArmorEnchants(Entity $entity, Event $event)
-                            $effect = Effect::getEffect(Effect::WEAKNESS);
-                            $effect->setAmplifier($enchantment->getLevel());
-                            $effect->setDuration(60 * $enchantment->getLevel());
-                            $effect->setVisible(false);
-                            $damager->addEffect($effect);
-                        }
-                        $enchantment = $armor->getEnchantment(CustomEnchantsIds::POISONED);
-                        if ($enchantment !== null && $damager->hasEffect(Effect::POISON) !== true) {
-                            $effect = Effect::getEffect(Effect::POISON);
-                            $effect->setAmplifier($enchantment->getLevel());
-                            $effect->setDuration(60 * $enchantment->getLevel());
-                            $effect->setVisible(false);
-                            $damager->addEffect($effect);
-                        }
-                        $enchantment = $armor->getEnchantment(CustomEnchantsIds::FROZEN);
-                        if ($enchantment !== null && $damager->hasEffect(Effect::SLOWNESS) !== true) {
-                            $effect = Effect::getEffect(Effect::SLOWNESS);
-                            $effect->setAmplifier($enchantment->getLevel());
-                            $effect->setDuration(60 * $enchantment->getLevel());
-                            $effect->setVisible(false);
-                            $damager->addEffect($effect);
-                        }
-                        $enchantment = $armor->getEnchantment(CustomEnchantsIds::REVULSION);
-                        if ($enchantment !== null && $damager->hasEffect(Effect::NAUSEA) !== true) {
-                            $effect = Effect::getEffect(Effect::NAUSEA);
-                            $effect->setAmplifier(0);
-                            $effect->setDuration(20 * $enchantment->getLevel());
-                            $effect->setVisible(false);
-                            $damager->addEffect($effect);
-                        }
-                        $enchantment = $armor->getEnchantment(CustomEnchantsIds::CURSED);
-                        if ($enchantment !== null && $damager->hasEffect(Effect::WITHER) !== true) {
-                            $effect = Effect::getEffect(Effect::WITHER);
-                            $effect->setAmplifier($enchantment->getLevel());
-                            $effect->setDuration(60 * $enchantment->getLevel());
-                            $effect->setVisible(false);
-                            $damager->addEffect($effect);
-                        }
-                        $enchantment = $armor->getEnchantment(CustomEnchantsIds::DRUNK);
-                        if ($enchantment !== null) {
-                            if (!$damager->hasEffect(Effect::SLOWNESS)) {
-                                $effect = Effect::getEffect(Effect::SLOWNESS);
+                    if ($damager instanceof Living) {
```

Comment:
Unnecessary check. In EntityDamageEvent, the damager is already checked to be a Player.

Reply:
This check fixed errors being reported by PHPStorm that `hasEffect()` and `addEffect()` were not found.  This is because `getDamager()` reports that it returns `Entity|null`.  Removed check and added a DocBlock which identifies `$damager` as `instance of Living` in:
https://github.com/DaPigGuy/PiggyCustomEnchants/pull/122/commits/69a8590d9a157ca0bedeaf0bb9d8130d321b9084